code/game/machinery/doors/windowdoor.dm

```diff
@@ -241,6 +241,13 @@
 	operating = FALSE
 	return 1
 
+///When the tram is in station, the doors are locked to engineering only.
+/obj/machinery/door/window/lock()
+	req_access = list("engineering")
```

Comment:
is this just to let them work on it? or for some other reason

Reply:
Yeah, so when the tram is in the station and doors are opened, we only want to allow Engineering staff to interact with it, so randoms don't just close the doors.